### ISSUE TITLE
Calibrated dual-replay priority exceeds 1.0 by one float32 ulp and permanently poisons the replay state

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#313]
## Summary

`DualReplayMemory` (`alberta_framework/core/dual_replay.py`) documents that the calibrated `*_weight` values "form a convex combination, so the resulting priority also lies in `[0, 1]`", and `_entries_are_valid` enforces `insertion_priorities <= 1.0`. But `_calibrated_priority` divides a numerator accumulated in float32 from *separately rounded* weights by a `weight_sum` rounded **once** from the float64 Python sum. With ordinary weights such as `(0.1, 0.2, 0.4)` and all three components saturated at `1.0` (coverage is `1.0` on the very first record because the long-term stratum is empty), the two roundings disagree and the priority comes out `1.0000001192092896`.

That value is written into `insertion_priorities` (short-term stratum unconditionally, long-term when admitted), so the state the module just produced fails its own range invariant. `record()` does not raise; every later `record`/`sample` returns `state_valid=False` and silently fails closed, `checkpoint_payload` raises, and the memory ingests nothing for the rest of the arm — a run that looks like "replay didn't help".

The default `(1.0, 1.0, 1.0)` is exact in float32, which is why the suite never sees it. Sampling 200k uniform weight triples, 13.6% produce a priority > 1.0; concrete already-normalized triples that trip it include `(0.27, 0.66, 0.07)` and `(0.33, 0.60, 0.07)`.

Same root, NaN variant: `_validate_config` validates the calibration scalars as float64 positives but they are consumed after a float32 cast — `float32(1e-46) == 0.0` gives `0/0 = NaN` for a legitimately zero surprise, and three `2e38` weights give an `inf` float32 weight sum, so `inf/inf = NaN` lands in `insertion_priorities` the same way.

## Falsifier (deterministic, pristine `main` @ `bae17ee`)

```python
import jax.numpy as jnp, jax.random as jr
from alberta_framework.core.dual_replay import (
    DualReplayConfig, DualReplayMemory, ReplayOutcome, ReplayPrediction)

cfg = DualReplayConfig(
    total_capacity=6, short_term_capacity=3, observation_dim=2, action_dim=2,
    short_term_sample_size=2, long_term_sample_size=2, long_term_policy="calibrated",
    surprise_scale=1.0, coverage_scale=1.0, progress_scale=1.0,
    surprise_weight=0.1, coverage_weight=0.2, progress_weight=0.4)
mem = DualReplayMemory(cfg)
f32 = lambda v: jnp.asarray(v, dtype=jnp.float32)
pred = ReplayPrediction(observation=f32([1., 2.]), action=f32([1., 1.]),
    old_behavior_probability=f32(0.25), old_behavior_probability_available=jnp.asarray(True),
    old_behavior_logit=f32(-1.25), old_behavior_logit_available=jnp.asarray(True),
    old_value_target=f32(1.), old_value_target_available=jnp.asarray(True),
    epistemic_surprise=f32(1.0), epistemic_surprise_available=jnp.asarray(True),
    aleatoric_uncertainty=f32(0.1), aleatoric_uncertainty_available=jnp.asarray(True),
    representation_version=jnp.asarray(0, jnp.int32), provenance_id=jnp.asarray(1, jnp.int32),
    source_id=jnp.asarray(7, jnp.int32), valid=jnp.asarray(True))
out = ReplayOutcome(next_observation=f32([2., 3.]), reward=f32(0.1), discount=f32(0.9),
    terminated=jnp.asarray(False), truncated=jnp.asarray(False),
    learning_progress=f32(1.0), learning_progress_available=jnp.asarray(True),
    safety_cost=f32(0.0), safety_cost_available=jnp.asarray(True), valid=jnp.asarray(True))

r1 = mem.record(mem.init(jr.key(0)), pred, out)
print(float(r1.long_term_priority), bool(mem.state_valid(r1.state)))
r2 = mem.record(r1.state, pred, out)
print(bool(r2.state_valid), bool(r2.wrote_short_term), int(r2.state.accepted_transition_count))
```

Output on `main`:

```
1.0000001192092896 False
False False 1
```

Expected: `1.0 True` then `True True 2` — the priority is closed in `[0, 1]` by construction, and calibration scalars that collapse to zero or overflow once narrowed to float32 are rejected at construction.

## Dedup

`gh issue/pr list --search` for "dual replay", "calibrated priority", "insertion_priorit", "convex combination" — no matches open or closed; no open PR touches `core/dual_replay.py`. Same class as the merged float32-domain validation sweep (#288/#289, #301) but on an untouched module.

Fix PR follows (failing-test-first).
